internal/signer/darwin/util/util.go

```diff
@@ -7,32 +7,54 @@ import (
 	"os"
 )
 
+const configsKey = "cert_configs"
+const macosKey = "macos_keychain"
+
 // EnterpriseCertificateConfig contains parameters for initializing signer.
 type EnterpriseCertificateConfig struct {
-	CertInfo CertInfo `json:"cert_info"`
+	CertInfo CertInfo
 }
 
 // CertInfo contains parameters describing the certificate to use.
 type CertInfo struct {
 	Issuer string `json:"issuer"`
 }
 
-// LoadCertInfo retrieves the certificate info from the config file.
-func LoadCertInfo(configFilePath string) (certInfo CertInfo, err error) {
+// LoadConfig retrieves the ECP config file.
+func LoadConfig(configFilePath string) (config EnterpriseCertificateConfig, err error) {
 	jsonFile, err := os.Open(configFilePath)
 	if err != nil {
-		return CertInfo{}, err
+		return EnterpriseCertificateConfig{}, err
 	}
 
 	byteValue, err := ioutil.ReadAll(jsonFile)
 	if err != nil {
-		return CertInfo{}, err
+		return EnterpriseCertificateConfig{}, err
 	}
-	var config EnterpriseCertificateConfig
-	err = json.Unmarshal(byteValue, &config)
+
+	var ecpConfig map[string]interface{}
+	err = json.Unmarshal(byteValue, &ecpConfig)
+
 	if err != nil {
-		return CertInfo{}, err
+		return EnterpriseCertificateConfig{}, err
+	}
+
+	for _, value := range ecpConfig[configsKey].([]interface{}) {
+		if v, ok := value.(map[string]interface{})[macosKey]; ok {
+			b, err := json.Marshal(v)
+
+			if err != nil {
+				return EnterpriseCertificateConfig{}, err
+			}
+
+			var certInfo CertInfo
+			err = json.Unmarshal(b, &certInfo)
+			if err != nil {
+				return EnterpriseCertificateConfig{}, err
+			}
+			return EnterpriseCertificateConfig{certInfo}, nil
+		}
 	}
-	return config.CertInfo, nil
 
+	return EnterpriseCertificateConfig{}, nil
```

Comment:
@andyrzhao Per discussion if there is no valid ECP config we will fallback to a non-mtls connection. Should I create a new error type to indicate that here? I'm unfamiliar with the prefered Go pattern